### PR TITLE
Enhance ONNX SmoothQuant tuning structure

### PR DESCRIPTION
**Pull request by yuwenzho**
## Type of Change

feature
API not change

## Description

Enhance SmoothQuant tuning structure and support alpha tuning for ONNX.
Details refer to PR [Enhance SmoothQuant tuning structure](https://github.com/intel/neural-compressor/pull/1109)

## Expected Behavior & Potential Risk

UT pass

## How has this PR been tested?

Local test & CI test

## Dependency Change?

no

### CODE REVIEW

File: test/algorithm/test_smooth_quant_onnx.py [hunk 1]
```diff
@@ -110,6 +158,89 @@ def test_sq(self):
             sq_tensor = numpy_helper.to_array(sq.model.get_initializer(init.name))
             self.assertAlmostEqual(tensor[0][0], sq_tensor[0][0], 4)
 
-
+    def _test_sq_tune_alpha_common(self, eval_func, alpha=np.arange(0.1, 0.2, 0.05).tolist(), quant_level=1):
+        from neural_compressor import quantization
+        from neural_compressor.config import PostTrainingQuantConfig, TuningCriterion
+        tuning_criterion = TuningCriterion(max_trials=8)
+
+        fp32_model = self.model
```
Comment (yiliu30) [inline test/algorithm/test_smooth_quant_onnx.py:166]:
```suggestion
        fp32_model = copy.deepcopy(self.model)
```

File: neural_compressor/adaptor/onnxrt.py [hunk 2]
```diff
@@ -186,29 +186,38 @@ def smooth_quant(self, model, dataloader, iterations, alpha=0.5, folding=True,
...
-        return {}
-        
+        """Check the model needs smooth quant or not."""
+        recipe_cfgs = tune_cfg.get('recipe_cfgs', None)
+        if recipe_cfgs and recipe_cfgs.get('smooth_quant', False) \
+            and recipe_cfgs['smooth_quant_args'].get('alpha', None):
+            # update alpha according to tune_cfg
+            self.cur_sq_args['alpha'] = \
+                tune_cfg['recipe_cfgs']['smooth_quant_args']['alpha']
+            return True
```
Comment (yiliu30) [inline neural_compressor/adaptor/onnxrt.py:218]: The `need_smooth_quant` condition should be: the `alpha` from `tune_cfg` is not `None` and not equal to the `alpha` in `cur_sq_args`.

Comment (yuwenzho) [reply neural_compressor/adaptor/onnxrt.py:218]: After alpha tuning, the best alpha might equal to the last tuned alpha.

For example: baseline acc = 1.0, alpha = [0.1, 0.2]
- alpha = 0.1, acc = 0.8 --> alpha = 0.2, acc = 0.9 -(end alpha tune)-> other tuning with best alpha = 0.2 which also needs excute smooth quant

File: neural_compressor/config.py [hunk 3]
```diff
@@ -847,7 +847,7 @@ def smooth_quant_args(val=None):
                         """
                         if isinstance(v, str):
                             assert v == "auto", "the alpha of sq only supports float and 'auto'"
```
Comment (mengniwang95) [inline neural_compressor/config.py:849]: the alpha of sq also support list

File: neural_compressor/adaptor/ox_utils/smooth_quant.py [hunk 4]
```diff
@@ -189,7 +195,7 @@ def recover(self):
                 scale = self.tensor_scales_info[key]
                 new_weight = weight * scale
                 self.model.set_initializer(input, new_weight)
-
+
```
Comment (mengniwang95) [inline neural_compressor/adaptor/ox_utils/smooth_quant.py:198]: remove blank

File: neural_compressor/adaptor/onnxrt.py (hunk 2 above)
Comment (yiliu30) [reply neural_compressor/adaptor/onnxrt.py:218]: I see, thx.

**Review by yiliu30 — APPROVED** [review]

**Review by xin3he — APPROVED** [review]

**Review by mengniwang95 — APPROVED** [review]

**Pull request merged**